### PR TITLE
fix: bundle externalValue with relative reference in examples object

### PR DESCRIPTION
**Pull request by AlexVarchuk**
## What/Why/How?
The bundle command doesn't correctly resolve `externalValue` with relative reference.

According to specification https://spec.openapis.org/oas/v3.1.0.html#fixed-fields-15 `externalValue` may contain relative references. Bundle command doesn't resolve it. 

## Reference

Resolves https://github.com/Redocly/redocly-cli/issues/919

## Testing
openapi.yaml
```yaml
openapi: 3.1.0
security: []
info:
  version: 1.0.0
  title: Example.com
  termsOfService: https://example.com/terms/
  contact:
    email: contact@example.com
    url: http://example.com/contact
  license:
    name: Apache 2.0
    url: http://www.apache.org/licenses/LICENSE-2.0.html
  description: OpenAPI description with external example
components: {}

paths:
  /:
    post:
      summary: Test request externalValue with relative reference in examples
      requestBody: 
        content:
          application/xml:
            schema:
              type: object
            examples:
              mergeRequest:
                summary: Merge request example
                externalValue: './external-value.json'
```

## Screenshots (optional)
Screenshot 2024-10-01 at 16 56 00

## Check yourself

- [ ] Code changed? - Tested with redoc/reference-docs/workflows (internal)
- [x] All new/updated code is covered with tests
- [ ] New package installed? - Tested in different environments (browser/node)

## Security

- [x] Security impact of change has been considered
- [x] Code follows company security practices and guidelines

### CODE REVIEW

File: packages/core/src/ref-utils.ts [hunk 1]
```diff
@@ -12,6 +12,10 @@ export function isRef(node: any): node is OasRef {
   return node && typeof node.$ref === 'string';
 }
 
+export function isRelativeReference(value: any): boolean {
```
Comment (tatomyr) [inline packages/core/src/ref-utils.ts:15]: Could you use the [existing check](https://github.com/Redocly/redocly-cli/blob/1f0ca52ce083f21bc1b39ec4f5da8a18ece5daf1/packages/core/src/ref-utils.ts#L69) instead?

File: packages/core/src/resolve.ts [hunk 2]
```diff
@@ -289,6 +297,14 @@ export async function resolveDocument(opts: {
         return;
       }
 
+      // handle externalValue with relative reference in examples object
+      if (node.externalValue && isRelativeReference(node.externalValue)) {
```
Comment (tatomyr) [inline packages/core/src/resolve.ts:301]: Wouldn't it be easier to treat `externalValue` as `$ref` in [isRef](https://github.com/Redocly/redocly-cli/blob/1f0ca52ce083f21bc1b39ec4f5da8a18ece5daf1/packages/core/src/ref-utils.ts#L11)?

File: .changeset/little-pears-sniff.md [hunk 3]
```diff
@@ -0,0 +1,6 @@
+---
+"@redocly/openapi-core": patch
+"@redocly/cli": patch
+---
+
+Fixed resolving `externalValue` with relative reference.
```
Comment (JLekawa) [inline .changeset/little-pears-sniff.md:6]:
```suggestion
Fixed an issue where the bundle command did not resolve relative links in `externalValue`.
```

**Review by JLekawa — APPROVED** [review]

File: .changeset/little-pears-sniff.md [hunk 4]
```diff
@@ -0,0 +1,6 @@
+---
+"@redocly/openapi-core": patch
+"@redocly/cli": patch
+---
+
+Fixed an issue where the bundle command did not resolve relative links in `externalValue`.
```
Comment (tatomyr) [inline .changeset/little-pears-sniff.md:6]:
```suggestion
Fixed an issue where the bundle command did not resolve links in `externalValue`.
```

**Review by tatomyr — APPROVED** [review]

File: packages/core/src/resolve.ts [hunk 5]
```diff
@@ -335,6 +335,28 @@ export async function resolveDocument(opts: {
         });
         resolvePromises.push(promise);
       }
+
+      // handle externalValue with relative reference in examples object
```
Comment (tatomyr) [inline packages/core/src/resolve.ts:339]:
```suggestion
      // Handle externalValue similar to $ref.
```

**Review by tatomyr — APPROVED** [review]
One small suggestion.

**Comment by tatomyr** [comment]
@AlexVarchuk also, please check it in Workflows.

**Comment by jeremyfiel** [comment]
This is for the `bundle` command but can it also be utilized for the `lint` command? 

My original issue was about linting

**Comment by tatomyr** [comment]
@jeremyfiel indeed. It needs to be further adjusted to be able to use it in linter, so I'll reopen the issue.

**Pull request merged**